File: backend/src/main/java/com/backend/backend/models/OrderItem.java
package com.backend.backend.models;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import lombok.Data;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Objects;

@Getter
@Setter
@Document(collection = "order_items")
public class OrderItem {
    @Id
    private String orderItemId;

    private Double price;
    private Integer quantity;

    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;
//    @DBRef
//    private Product product;
    @DBRef
    private Customization customization;

}

File: frontend/src/models/OrderItem.js
// OrderItem.js
export class OrderItem {
    constructor(orderItemId, order, product, customization, price, mergedImage, quantity) {
        this.orderItemId = orderItemId;
        this.order = order;           // Reference to the Order object
        this.product = product;       // Reference to the Product object
        this.customization = customization || null; // Customization can be null
        // this.mergedImage = mergedImage;
        this.price = price;
        this.quantity = quantity;
    }
}

// module.exports = OrderItem;
